frontend/taipy-gui/src/components/Taipy/TaipyStyle.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React, { ReactNode } from "react";
+import { Helmet } from "react-helmet-async";
+
+interface TaipyStyleProps {
+    className: string;
+    content: string;
+}
+
+export const getComponentClassName = (children: ReactNode) =>
+    (
+        React.Children.map(children, (element) =>
+            React.isValidElement(element) && element.type == TaipyStyle ? element.props.className : undefined
+        )?.filter((v) => v) || []
+    ).join(" ");
+
+const getStyle = (style: Record<string, unknown>): string =>
+    Object.entries(style)
+        .map(
+            ([k, v]) =>
+                `${k}{${
+                    typeof v == "string"
+                        ? v
+                        : Object.entries(v as Record<string, unknown>)
```

Comment:
No test if Object? I guess that's guaranteed by the back-end implementation.

Reply:
You're right I'll add that
Python doesn't care

Reply:
You're right I'll add that
Python doesn't care